// Copyright 2022 Adap GmbH. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// ==============================================================================

syntax = "proto3";

package flwr.proto;

import "flwr/proto/node.proto";
import "flwr/proto/transport.proto";

message Task {
  Node producer = 1;
  Node consumer = 2;
  string created_at = 3;
  string delivered_at = 4;
  string ttl = 5;
  repeated string ancestry = 6;

  ServerMessage legacy_server_message = 101 [ deprecated = true ];
  ClientMessage legacy_client_message = 102 [ deprecated = true ];
}

message TaskIns {
  string task_id = 1;
  string group_id = 2;
  string workload_id = 3;
  Task task = 4;
}

message TaskRes {
  string task_id = 1;
  string group_id = 2;
  string workload_id = 3;
  Task task = 4;
}
